File: sdk/node/botlaunch.mjs
// Minimal BotLaunch API client — zero dependencies (Node 18+ global fetch).
// Usage: const bl = new BotLaunch(process.env.BOTLAUNCH_API_KEY);
const BASE = 'https://api.botlaunch.io/api/v1';

export class BotLaunchError extends Error {
  constructor(status, body) {
    super(body?.error || body?.message || `HTTP ${status}`);
    this.status = status;
    this.body = body;
  }
}

export class BotLaunch {
  constructor(apiKey, baseUrl = BASE) {
    if (!apiKey || !apiKey.startsWith('bl_live_')) {
      throw new Error('A BotLaunch API key (bl_live_...) is required.');
    }
    this.apiKey = apiKey;
    this.baseUrl = baseUrl;
  }

  async request(method, path, body) {
    const res = await fetch(`${this.baseUrl}${path}`, {
      method,
      headers: {
        Authorization: `Bearer ${this.apiKey}`,
        'Content-Type': 'application/json',
      },
      body: body ? JSON.stringify(body) : undefined,
    });
    const json = await res.json().catch(() => ({}));
    if (!res.ok) throw new BotLaunchError(res.status, json);
    return json.data ?? json;
  }

  context() { return this.request('GET', '/context'); }
  listBots() { return this.request('GET', '/bots'); }
  validateToken(token) { return this.request('POST', '/bots/validate-token', { token }); }
  connectBot(payload) { return this.request('POST', '/bots', payload); }
  botStatus(id) { return this.request('GET', `/bots/${id}/status`); }
  moduleDefinitions() { return this.request('GET', '/bot-modules/definitions'); }
  setModule(groupId, type, enabled, config) {
    return this.request('PATCH', `/bot-modules/groups/${groupId}/${type}`, { enabled, config });
  }
  listDesigns() { return this.request('GET', '/commerce/designs'); }
  createProduct(groupId, product) { return this.request('POST', `/groups/${groupId}/products`, product); }
}

File: sdk/python/botlaunch.py
"""Minimal BotLaunch API client (requires `requests`)."""
import requests

BASE = "https://api.botlaunch.io/api/v1"


class BotLaunchError(Exception):
    def __init__(self, status, body):
        super().__init__(body.get("error") or body.get("message") or f"HTTP {status}")
        self.status = status
        self.body = body


class BotLaunch:
    def __init__(self, api_key: str, base_url: str = BASE):
        if not api_key or not api_key.startswith("bl_live_"):
            raise ValueError("A BotLaunch API key (bl_live_...) is required.")
        self.api_key = api_key
        self.base_url = base_url

    def _req(self, method: str, path: str, body=None):
        r = requests.request(
            method, f"{self.base_url}{path}",
            headers={"Authorization": f"Bearer {self.api_key}", "Content-Type": "application/json"},
            json=body, timeout=30,
        )
        data = r.json() if r.content else {}
        if not r.ok:
            raise BotLaunchError(r.status_code, data)
        return data.get("data", data)

    def context(self): return self._req("GET", "/context")
    def list_bots(self): return self._req("GET", "/bots")
    def validate_token(self, token): return self._req("POST", "/bots/validate-token", {"token": token})
    def module_definitions(self): return self._req("GET", "/bot-modules/definitions")
    def set_module(self, group_id, mtype, enabled, config):
        return self._req("PATCH", f"/bot-modules/groups/{group_id}/{mtype}", {"enabled": enabled, "config": config})
    def create_product(self, group_id, product): return self._req("POST", f"/groups/{group_id}/products", product)
